### ISSUE TITLE
majestic failed after attempt to connect from Onvifer

### ISSUE THREAD

**Issue by mariarti** [OpenIPC/majestic_issues#1]
camera: IVG85HG50PYA-S
openipc.hi3516ev300-br.tgz
Commit: d9d0626 
Branch: master 
Tag: latest 

Clean install by this man: https://github.com/OpenIPC/openipc-2.1/wiki/install_hisi
default /etc/majestic.yaml
steps to reproduce:
1) open in browser http://ip:8888/
    video runs good
2) open stream 
    $ mpv rtsp://ip:554/stream=0
     video runs good
3) open same rtsp://ip:554/stream=0 in android app Onvifer 15.27 (manual settings with transport protocol: TCP)
  Fail
  
what happened:
majestic gone away (pid=781)
```
# cat /dev/kmsg
3,181,132920576,-;mmz_userdev:mmz_userdev_release: 
0,182,132920598,-;MMB LEAK(pid=781): 0x45B80000, 249856 bytes, 'AENC(0)StrmBuf'
4,183,132920604,-;mmz_userdev:mmz_userdev_release: 
0,184,132920609,-;mmb<0x45b80000> mapped to userspace 0xb5d32000 will be unmaped!
3,185,132920667,-;mmz_userdev:mmz_userdev_release: 
0,186,132920673,-;MMB LEAK(pid=781): 0x420D4000, 16384 bytes, 'AENC(0) CirBuf'
4,187,132920675,-;mmz_userdev:mmz_userdev_release: 
0,188,132920679,-;mmb<0x420d4000> mapped to userspace 0xb5d2e000 will be unmaped!
```
Expectation:
WDT should up majestic.
During boot it was initialized, but seems don't work.
```
Hisilicon Watchdog Timer: 0.01 initialized. default_margin=60 sec (nodeamon= 0)
```
Reboot helps until Onvifer 15.27 will be launched.

Also, after reboot majestic comes up and running and after single connection from Onvifer 15.27 same gone away.

**Comment by widgetii** [comment]
Please also show `free` on your system

**Comment by mariarti** [comment]
```
# free
              total        used        free      shared  buff/cache   available
Mem:         124224       85692       28600          96        9932       36980
Swap:             0           0           0
```

**Comment by mariarti** [comment]
```
root@ipcam:~# killall majestic
root@ipcam:~# ulimit -c unlimited
root@ipcam:~# echo /tmp/%e.%p.%s.%t.core > /proc/sys/kernel/core_pattern
root@ipcam:~# majestic
17:37:03 [    main] main@143                     Majestic version master+5b7fec4, built on 2021-04-27
17:37:03 [app_conf] load_config@152              Using /etc/majestic.yaml as main configuration
17:37:03 [app_conf] parse_app_config@237         Autodetected sensor as 'imx335_i2c'
17:37:03 [app_conf] find_sensor_config@115       matched sensor config: imx335_i2c_4M_line.ini
17:37:03 [app_conf] find_sensor_config@125       Using /etc/sensors/imx335_i2c_4M_line.ini as sensor configuration
17:37:03 [  config] parse_int64@416              Can't parse param 'volume' value 'auto'. Is not a integer (dec or hex) number.
17:37:03 [    main] main@160                     app_config.max_pool_cnt 128
17:37:03 [    main] main@161                     app_config.blk_cnt 4
17:37:03 [   hisdk] start_sdk@1546               App was built with MPP version: Hi3516EV200_MPP_V1.0.1.2.B030 Release
17:37:03 [   hisdk] start_sdk@1549               Current MPP version: HI_VERSION=Hi3516EV200_MPP_V1.0.1.2 B030 Release
17:37:03 [   hisdk] start_sdk@1559               chipid 3516E300
17:37:03 [   hisdk] start_sdk@1567               sensor_config.sensor_type stSnsImx335Obj
17:37:03 [   hisdk] start_sdk@1568               sensor_config.dll_file /usr/lib/sensors/libsns_imx335.so
17:37:03 [   hisdk] start_sdk@1569               sensor_config.dev_attr 2
17:37:03 [   hisdk] start_sdk@1571               sensor_config.vichn.pix_format 26
17:37:03 [   hisdk] start_sdk@1573               sensor_config.input_mode 0
17:37:03 [   hisdk] start_sdk@1574               sensor_config.mode 0
17:37:03 [   hisdk] start_sdk@1577               sensor_config.isp.isp_x 0
17:37:03 [   hisdk] start_sdk@1578               sensor_config.isp.isp_y 0
17:37:03 [   hisdk] start_sdk@1579               sensor_config.isp.isp_w 2592
17:37:03 [   hisdk] start_sdk@1580               sensor_config.isp.isp_h 1520
17:37:03 [   hisdk] start_sdk@1583               sensor_config.isp.isp_frame_rate 30
17:37:03 [   hisdk] start_sdk@1584               sensor_config.isp.isp_bayer 0
17:37:03 [  sensor] tryLoadLibrary@23            try to load: /usr/lib/sensors/libsns_imx335.so
17:37:03 [  sensor] tryLoadLibrary@25            libsns_so 0x00000000B6D24880
17:37:03 [   hisdk] free_mem@779                 Free MMZ mem before allocation: 98272KB
17:37:03 [   hisdk] init_mem@1676                u32AlignWidth: 64
17:37:03 [   hisdk] dump_vb_configuration@1651   VB configuration:
17:37:03 [   hisdk] dump_vb_configuration@1661     [0]: 5982720 x 7
17:37:03 [   hisdk] dump_vb_configuration@1661     [1]: 8375807 x 1
17:37:03 [   hisdk] dump_vb_configuration@1661     [4]: 345600 x 1
17:37:03 [   hisdk] free_mem@779                 Free MMZ mem after allocation: 48840KB
17:37:03 [   hisdk] init_sensor@1846             Sensor driver has been loaded
linear mode
binit false IMX335 i2c init
-------Sony IMX335_init_5M_2592x1944_12bit_linear30 Initial OK!-------
17:37:03 [   hisdk] init_vpss@2287               HI_MPI_SYS_Bind: VIU(0)->VPSS(grp 0)
17:37:03 [   hisdk] log_venc_chn@885             H.265 vpss_chn(0)->venc_chn(0)   2592x1520 25fps 4096Kbit
17:37:03 [   hisdk] create_vpss_chn@586            new venc: 0   vpss_grp: 0,   vpss_chn: 0
17:37:03 [   hisdk] create_vpss_chn@665            create_venc_chn  VPSS(0)->VENC(0)
17:37:03 [   hisdk] start_video@1272               venc took 15304KB of MMZ [2885]
17:37:03 [   hisdk] start_jpeg@794               JPEG snapshot venc_chn 1  2592x1520
17:37:03 [   hisdk] bind_vpss_venc@698             bind_vpss_venc  VPSS(0)->VENC(1)
17:37:03 [   hisdk] start_jpeg@876                 jpeg_enc took 3856KB of MMZ [3847]
17:37:03 [   audio] init_audio_in@592            Ai(0,0) bind to AencChn:0 ok!
17:37:03 [   hisdk] start_sdk@1636                 Audio took 512KB of MMZ
17:37:03 [   hisdk] start_sdk@1643               HiSilicon SDK has been started.
17:37:03 [   hisdk] free_mem@779                 Free MMZ mem finally: 22868KB
17:37:03 [image_pa] image_params_tuning@45       Image tuning thread has been started
17:37:03 [  evhttp] start_web_server@134         Using LAME v3.100
17:37:03 [  evhttp] start_web_server@194         HTTP server has been started on 0.0.0.0:8888.
17:37:03 [    rtsp] rtsp_init@143                RTSP server started on port 554
17:37:07 [   hisdk] set_contrast@2667            Autoadjust image param cscAttrContrField from 50 to 90
╮                                     ╭╮                                             
│                                     │╰╮                                            
│                                    ╭╯ │                                            
│                                   ╭╯  ╰╮                                           
│                                ╭──╯    │                                           
│                                │       ╰╮                                          
│                             ╭─╮│        ╰╮                                         
│                             │ ╰╯         │                                         
│                            ╭╯            │                                         
│                            │             │                                         
│                           ╭╯             │                                         
│                        ╭──╯              │                                         
│                        │                 │                                         
│                        │                 ╰╮                                        
│                       ╭╯        ╭─╮ ╭╮    ╰╮ ╭──╮  ╭─╮                             
│           ╭╮         ╭╯  ╭╮    ╭╯ ╰─╯│╭╮   ╰─╯  ╰╮ │ │                             
│           ││        ╭─╮ ╭╯╰────╯     ╰╯│         ╰─╯ │                             
│           ││      ╭─╯╯╰─╯              ╰╮            │                             
│          ╭╯│    ╭─╯─╯                   ╰╮           ╰╮                            
│         ╭╯╭─────╯╭╯                      ╰╮╭╮         ╰───╮ ╭─╮╮                   
│       ╭╮│╭╯ │    │                        ╰╯╰───────────────╯ ╰╮╮                  
│    ╭──╯╰─╯  ╰────╯                                             ╰╮╮           ╭──╮  
╰────╯    │                                                       ╰────────────╯  ╰──
──────────╯                                                          ╰───────────────





17:37:58 [    rtsp] listener_cb@118              New RTSP client from: 192.168.1.245:46252
17:37:58 [rtsp_cli] RTSPClient_dispatch@239      RTSP DESCRIBE rtsp://192.168.1.173:554/stream=0 CSeq=0 from 192.168.1.245:46252
17:37:58 [rtsp_cha] RTSPChannel_describe_sdp_video@263 RTSP chn 0 sps_len 0
17:37:58 [rtsp_cha] RTSPChannel_describe_sdp_video@266 RTSP chn 0 pps_len 0
17:37:58 [rtsp_cha] RTSPChannel_describe_sdp_video@273 RTSP chn 0 profile_level_id 000000
17:37:58 [rtsp_cli] RTSPClient_dispatch@239      RTSP SETUP rtsp://192.168.1.173:554/stream=0//video CSeq=1 from 192.168.1.245:46252
17:37:58 [rtsp_cha] RTSPChannel_setup@182        TCP RTSP SETUP done, stream_id=0 from 192.168.1.245:46252
17:37:59 [rtsp_cli] RTSPClient_dispatch@239      RTSP PLAY rtsp://192.168.1.173:554/stream=0 CSeq=2 from 192.168.1.245:46252
Segmentation fault (core dumped)
```

**Comment by mariarti** [comment]
[majestic.806.11.1619707881.zip](https://github.com/OpenIPC/majestic_issues/files/6399576/majestic.806.11.1619707881.zip)

**Comment by ZigFisher** [comment]
> camera: IVG85HG50PYA-S
> openipc.hi3516ev300-br.tgz
> Commit: d9d0626
> Branch: master
> Tag: latest

I have Onvifer 15.27 and the same problem with performance.
Try disabling the audio section in the configuration file.
And do the testing again. Thanks.

**Comment by mariarti** [comment]
> > camera: IVG85HG50PYA-S
> > openipc.hi3516ev300-br.tgz
> > Commit: d9d0626
> > Branch: master
> > Tag: latest
> 
> I have Onvifer 15.27 and the same problem with performance.
> Try disabling the audio section in the configuration file.
> And do the testing again. Thanks.

Confirm, after disabling audio in /etc/majestic.yaml video still works after connect from Onvifer.

Thanks @ZigFisher for quick workaround!

**Comment by widgetii** [comment]
Fixed in `version master+1749648, built on 2021-05-13`
